File: THPT_LUONG_TAI/contestants/nguyen_duy_hung/DGROUP.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    ifstream inp("DGROUP.INP");
    ofstream out("DGROUP.OUT");
    int a,b,c;
    inp >> a >> b >> c;
    int k=0;
    for(int j=1;j<=b;j++)
    {
        if(a-j*2+b-j>=6 && a>=j*2 && b>=j)
        {
            k=j;
        }
    }
    out << k;
    out.close();
    return 0;

}

File: THPT_LUONG_TAI/contestants/bach_do/dgroup.py
with open("dgroup.inp") as fi:
    n,m,k=map(int,fi.readline().split()) 
    a=0 
with open("dgroup.out") as fo:
    if n//2 > m:
        a+=m
        k-=(n//2-m+n%2)
    elif n//2 < m:
        a+=n//2
        k-=(m-n//2+n%2)
    elif n//2==m:
        a+=m
    if k<=0:
        fo.write(str(a))
    elif k>0:
        while k>0:
            a-=1
            k-=6
        fo.write(str(a))
fi.write()
fo.close()       